actions/ci-beholder-validator/README.md

```diff
@@ -0,0 +1,278 @@
+roles and permissions. A sample PR can be found
+[here](https://github.com/smartcontractkit/infra/pull/6974/files) Once the above
+PR is run, you will have access to the `{{secrets.aws-role-arn}}`.
+`aws-account-number` is used to fetch the image from the ECR. Hence this is same
+as the aws account id of production account. The secrets can be inserted in the
+repo secrets by following guide
+[here](https://smartcontract-it.atlassian.net/wiki/spaces/RE/pages/906985607/GitHub+Repo+Configuration#GitHub-Secrets)
+
+### Step 3: Open a Pull Request
+
+When you push changes to schemas/pet.proto in a feature branch and open a PR,
+GitHub Actions will automatically:
+
+1. Spin up Redpanda for testing
+2. Pull and run the schema-validator container
+3. Compare changes in pet.proto to what’s in the main branch
+4. Fail if there’s a backward-incompatible change
+5. If validation passes, you’ll see a success status in your PR checks.
+
+---
+
+## Configuration
+
+### Required Inputs
+
+- docker-registry: Registry to pull the validator image from (currently 'aws').
+
+### Optional AWS Inputs
+
+If docker-registry: 'aws', you can supply:
+
+- aws-region: AWS region hosting ECR (e.g., us-west-2).
+- aws-role-arn: IAM Role ARN to assume for ECR.
+- aws-role-duration-seconds: Session duration (default: 900).
+- aws-account-number: AWS account ID (required if your ECR is in a non-default
+  account).
+
+---
+
+## Schema Configuration (`beholder.yaml`)
+
+The `beholder.yaml` file describes your schemas and their locations. Here’s an
+example:
+
+```yaml
+beholder:
+  domain: your_domain
+  schemas:
+    - entity: UserEvent
+      schema: "./schemas/user_event.proto"
+    - entity: OnRampEvent
+      schema: "./schemas/on_ramp_event.avsc"
+```
+
+- domain: Logical namespace for your schemas, e.g., payment_service.
+- schemas: List of entities, each with a friendly entity name and schema path.
+- Paths such as schema: "./schemas/user_event.proto" must match your actual
+  folder structure.
+
+## Troubleshooting
+
+1. AWS Authentication Failed
+   - Check if aws-role-arn is correct.
+   - Ensure the role has ECR pull permissions.
+   - Verify aws-region matches your ECR location.
+2. Schema Validation Failed
+   - Make sure your changes are backward-compatible.
+   - Verify schema file paths in beholder.yaml match the actual files.
+   - Check logs for parse or registry errors.
+3. Docker Issues
+   - The runner must have Docker installed (e.g., ubuntu-latest includes it).
+   - Check network connectivity to ECR or your registry.
+   - If you see “pull access denied,” confirm you’re logged in or have correct
+     permissions.
+4. File Not Detected
+   - Verify your PR modifies .proto, .avsc, or a path included under
+     on.pull_request.paths.
+   - Ensure beholder.yaml is in the correct location.
+
+## Contributing
+
+Contributions are welcome! Please submit PRs with improvements or bug fixes.
```

Comment:
Is this going to be used by people external to CL? If yes, then the requirement of ECR permissions will not work.

Reply:
This is removed.